libs/libc/semaphore/sem_getvalue.c

```diff
@@ -65,7 +69,23 @@ int nxsem_get_value(FAR sem_t *sem, FAR int *sval)
 {
   if (sem != NULL && sval != NULL)
     {
-      *sval = atomic_read(NXSEM_COUNT(sem));
+      if (NXSEM_IS_MUTEX(sem))
+        {
+          uint32_t mholder = atomic_read(NXSEM_MHOLDER(sem));
+          if (NXSEM_MACQUIRED(mholder))
+            {
+              *sval = NXSEM_MBLOCKS(mholder) ? -1 : 0;
```

Comment:
Could this be simplified to:

```
uint32_t mholder = atomic_read(NXSEM_MHOLDER(sem));
if (mholder == NXSEM_NO_MHOLDER)
  {
    *sval = 1;
  }
else
  {
    *sval = 0;
  }
}
```

Reply:
Yes I agree. This is a leftovers from nxmutex being based on counting semaphore, I'll remove that. nxsem_getvalue for a mutex should just return -EINVAL.